charts/argo-cd/templates/argocd-server-service.yaml

```diff
@@ -13,6 +13,7 @@ metadata:
   annotations:
 {{ toYaml .Values.server.serviceAnnotations | indent 4}}{{- end }}
 spec:
+  type: {{ .Values.server.serviceType }}
```

Comment:
I think you need if condition around this new field? Or is `ClusterIP` default value?

Reply:
ClusterIP is the default value: https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types.

However, if its preferable I could wrap this in a conditional and leave ClusterIP out of the values file. That way it's still modifiable, but if not added it will have the exact same functionality it had before the change.